plugin/luadev.vim

```diff
@@ -1,23 +1,47 @@
 command! Luadev lua require'luadev'.start()
 
 noremap <Plug>(Luadev-RunLine) <Cmd>lua require'luadev'.exec(vim.api.nvim_get_current_line())<cr>
-vnoremap <silent> <Plug>(Luadev-Run) :<c-u>call luaeval("require'luadev'.exec(_A)", <SID>get_visual_selection())<cr>
+vnoremap <silent> <Plug>(Luadev-Run) :<c-u>call <SID>luadev_run_operator('visual')<cr>
+noremap <silent> <Plug>(Luadev-RunOperator) :<c-u>set opfunc=<SID>luadev_run_operator<cr>g@
 noremap <silent> <Plug>(Luadev-RunWord) :<c-u>call luaeval("require'luadev'.exec(_A)", <SID>get_current_word())<cr>
 inoremap <Plug>(Luadev-Complete) <Cmd>lua require'luadev.complete'()<cr>
 
+
 " thanks to @xolox on stackoverflow
-function! s:get_visual_selection()
-    let [lnum1, col1] = getpos("'<")[1:2]
-    let [lnum2, col2] = getpos("'>")[1:2]
+" same function for visual and normal. Except visual passes a parameter.
+function! s:luadev_run_operator(type = '')
```

Comment:
This could be made a lot less verbose, c f https://github.com/bfredl/nvim-ipy/blob/c21d8a78e209546ff6649e990ce71fcfa89baa14/plugin/ipy.vim#L36 :)

Reply:
Oh thanks, I changed a little bit since operator passes a string and is falsy. So no extra function function. Hope is fine 😄